File: Java/TP2-2/src/Cliente.java

public class Cliente {

	String nome;
	private double saldo;
	private double limite;

	public Cliente(String nome, double limite) {
		this.nome = nome;
		this.limite = limite;
	}

	public void depositar(double valor) {
		this.saldo += valor;
	}

	public void sacar(double valor) {
		if (valor <= saldo + limite) {
			saldo -= valor;
		}
	}

	public double checarSaldo() {
		return saldo + limite;
	}

	public String getNome() {
		return nome;
	}

	public double getSaldo() {
		return saldo;
	}

	public double getLimite() {
		return limite;
	}

}

File: Kotlin/TPK2-2/bin/Cliente.kt
class Cliente(nome:String, limite:Double) {
  var nome:String
  internal set
  var saldo:Double = 0.toDouble()
  var limite:Double = 0.toDouble()
  init{
    this.nome = nome
    this.limite = limite
  }
  fun depositar(valor:Double) {
    this.saldo += valor
  }
  fun sacar(valor:Double) {
    if (valor <= saldo + limite)
    {
      saldo -= valor
    }
  }
  fun checarSaldo():Double {
    return saldo + limite
  }
}